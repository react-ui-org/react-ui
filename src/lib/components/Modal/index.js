export { default as Modal } from './Modal';
export { default as ModalBody } from './ModalBody';
export { default as ModalCloseButton } from './ModalCloseButton';
export { default as ModalContent } from './ModalContent';
export { default as ModalHead } from './ModalHead';
export { default as ModalFooter } from './ModalFooter';
