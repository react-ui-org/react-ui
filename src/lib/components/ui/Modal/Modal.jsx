import PropTypes from 'prop-types';
import React from 'react';
import { createPortal } from 'react-dom';
import Button from '../Button';
import {
  Toolbar,
  ToolbarItem,
} from '../../layout/Toolbar';
import transferProps from '../../../utils/transferProps';
import { withTranslationContext } from '../../../translation';
import styles from './Modal.scss';

class Modal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isContentOverflowing: false,
    };

    this.setGradient = this.setGradient.bind(this);
    this.pressEscapeHandler = this.pressEscapeHandler.bind(this);
  }

  componentDidMount() {
    window.document.addEventListener('keydown', this.pressEscapeHandler, false);
  }

  componentWillUnmount() {
    window.document.removeEventListener('keydown', this.pressEscapeHandler, false);
  }

  setGradient() {
    if (!this.state.isContentOverflowing) {
      this.setState({ isContentOverflowing: true });
    }
  }

  pressEscapeHandler(e) {
    if (e.keyCode === 27 && this.props.closeHandler) {
      this.props.closeHandler();
    }
  }

  preRender() {
    const sizeClass = (size) => {
      if (size === 'small') {
        return styles.isRootSmall;
      }

      if (size === 'medium') {
        return styles.isRootMedium;
      }

      if (size === 'large') {
        return styles.isRootLarge;
      }

      return styles.isRootAuto;
    };

    return (
      <div
        className={styles.overlay}
        id={this.props.id}
        onClick={(e) => {
          if (this.props.closeHandler) {
            this.props.closeHandler(e);
          }
        }}
        onScroll={this.setGradient}
        role="presentation"
      >
        <div
          className={`
            ${styles.root}
            ${sizeClass(this.props.size)}
            ${this.state.isContentOverflowing ? styles.isContentOverflowing : ''}
          `.trim()}
          onClick={(e) => {
            e.stopPropagation();
          }}
          role="presentation"
        >
          <div className={styles.head}>
            <h3
              className={styles.headTitle}
              {...(this.props.id && { id: `${this.props.id}__title` })}
            >
              {this.props.title}
            </h3>
            {this.props.closeHandler && (
              <button
                type="button"
                className={styles.close}
                onClick={this.props.closeHandler}
                title={this.props.translations.close}
                {...(this.props.id && { id: `${this.props.id}__closeModalHeaderButton` })}
              >
                Ã—
              </button>
            )}
          </div>
          <div
            className={styles.body}
            {...(this.props.id && { id: `${this.props.id}__content` })}
          >
            {this.props.children}
          </div>
          <div className={styles.footer}>
            <Toolbar dense>
              {this.props.actions.map((action) => (
                <ToolbarItem key={action.label}>
                  <Button
                    {...transferProps(action)}
                    clickHandler={action.clickHandler}
                    disabled={action.disabled}
                    id={action.id || undefined}
                    label={action.label}
                    loadingIcon={action.loadingIcon}
                    variant={action.variant}
                  />
                </ToolbarItem>
              ))}
              {this.props.closeHandler && (
                <ToolbarItem>
                  <Button
                    clickHandler={this.props.closeHandler}
                    label={this.props.translations.close}
                    priority="flat"
                    {...(this.props.id && { id: `${this.props.id}__closeModalFooterButton` })}
                  />
                </ToolbarItem>
              )}
            </Toolbar>
          </div>
        </div>
      </div>
    );
  }

  render() {
    const { portalId } = this.props;

    if (portalId === null) {
      return this.preRender();
    }

    return createPortal(this.preRender(), document.getElementById(portalId));
  }
}

Modal.defaultProps = {
  actions: [],
  closeHandler: null,
  id: undefined,
  portalId: null,
  size: 'medium',
};

Modal.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.shape({
    clickHandler: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    id: PropTypes.string,
    label: PropTypes.string.isRequired,
    loadingIcon: PropTypes.node,
    variant: PropTypes.string,
  })),
  children: PropTypes.node.isRequired,
  closeHandler: PropTypes.func,
  id: PropTypes.string,
  portalId: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large', 'auto']),
  title: PropTypes.string.isRequired,
  translations: PropTypes.shape({
    close: PropTypes.string.isRequired,
  }).isRequired,
};

export default withTranslationContext(Modal, 'Modal');
