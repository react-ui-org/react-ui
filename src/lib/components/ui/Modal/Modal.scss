@import '../../../styles/tools/offset';
@import 'theme';
@import 'variables';

.root {
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: $modal-zindex;
  display: flex;
  flex-direction: column;
  max-width: calc(100% - (2 * #{$modal-padding-x}));
  max-height: calc(100% - (2 * #{$modal-padding-y}));
  overflow-y: auto;
  border-radius: $modal-border-radius;
  background-color: $modal-background;
  box-shadow: $modal-box-shadow;
  transform: translate(-50%, -50%);
}

.head {
  display: flex;
  flex: none;
  align-items: baseline;
  justify-content: space-between;
  padding: $modal-padding-y offset(3) $modal-padding-y $modal-padding-x;
}

.headTitle {
  margin-bottom: 0;
  font-size: $modal-head-title-font-size;
}

.close {
  padding: offset(1) offset(2);
  font-size: map-get($typography-size-values, 3);
  line-height: 1;
  color: inherit;
  border: 0;
  background: none;
  box-shadow: none;
  appearance: none;
  user-select: none;
  cursor: pointer;
}

.body {
  padding: $modal-padding-y $modal-padding-x;
  overflow-y: auto;
}

.footer {
  display: flex;
  flex: none;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  padding: $modal-padding-y $modal-padding-x;
  border-bottom-right-radius: $modal-border-radius;
  border-bottom-left-radius: $modal-border-radius;
  background-color: $modal-footer-background;
}

.overlay {
  position: fixed;
  top: 0;
  left: 0;
  z-index: $modal-overlay-zindex;
  width: 100vw;
  height: 100vh;
  background-color: $modal-overlay-background;
}

.isContentOverflowing .head {
  box-shadow: $modal-head-box-shadow;
}

.isContentOverflowing .footer {
  box-shadow: $modal-footer-box-shadow;
}

.isRootSmall {
  width: $modal-size-small-width;
}

.isRootMedium {
  width: $modal-size-medium-width;
}

.isRootLarge {
  width: $modal-size-large-width;
}

.isRootAuto {
  width: auto;
}
