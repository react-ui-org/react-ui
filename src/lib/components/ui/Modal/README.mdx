---
name: Modal
menu: 'Components: UI'
route: /components/ui/modal
---

# Modal

Modal allows to prompt users to take or complete an action.

import {
  Playground,
  Props,
} from 'docz'
import { Button } from '../Button/Button'
import {
  TranslatedModal as Modal,
  Modal as ParsableModal,
} from './Modal'

## Basic Usage

To implement the Modal component, you need to import it first:

```js
import { Modal } from '@react-ui-org/react-ui';
```

And use it:

<Playground>
  {() => {
    const [modalOpen, setModalOpen] = React.useState(false);
    return (
      <>
        <Button
          clickHandler={() => setModalOpen(true)}
          label="Launch modal"
          priority="outline"
        />
        <div>
          {modalOpen && (
            <Modal
              title="Warning"
              actions={[{
                clickHandler: () => setModalOpen(false),
                label: 'Delete',
                variant: 'danger',
              }]}
              closeHandler={() => setModalOpen(false)}
            >
              <p>
                Do you really want to delete the user <code>admin</code>?
                This cannot be undone.
              </p>
            </Modal>
          )}
        </div>
      </>
    );
  }}
</Playground>

See [API](#api) for all available options.

## General Guidelines

- Use modals to **confirm an action,** display a **blocking alert**, or to
  reveal **contextual options or settings** that cannot be displayed inline
  with the parent content.

- **The title** should communicate the **purpose of the modal** rather than a
  generic text. Eg. ‚ÄúDelete the user?‚Äù tells more than ‚ÄúAre you sure?‚Äù or
  ‚ÄúWarning‚Äù.

- **Modal actions** should correspond to the modal purpose, too. Eg. ‚ÄúDelete‚Äù
  tells better what happens rather than ‚ÄúOK‚Äù.

- Modal **automatically focuses the first form field** by default which allows
  users to confirm the modal by hitting enter key (requires an action of type
  `submit`). When no field is found then the first button in footer is focused.
  To turn this feature off, set the `autofocus` prop to `false`.

- **Avoid stacking** of modals. While it may technically work, the modal is just
  not designed for that.

## Sizes

Modal is available in three fixed-width sizes: small, medium, and large. Modals
of any size automatically shrink when they cannot fit the screen width.

<Playground>
  {() => {
    const [modalOpen, setModalOpen] = React.useState(false);
    const [modalSize, setModalSize] = React.useState('small');
    return (
      <>
        <Button
          clickHandler={() => {
            setModalSize('small');
            setModalOpen(true);
          }}
          label="Launch small modal"
          priority="outline"
        />
        <Button
          clickHandler={() => {
            setModalSize('medium');
            setModalOpen(true);
          }}
          label="Launch medium modal"
          priority="outline"
        />
        <Button
          clickHandler={() => {
            setModalSize('large');
            setModalOpen(true);
          }}
          label="Launch large modal"
          priority="outline"
        />
        <div>
          {modalOpen && (
            <Modal
              title="Delete the user?"
              actions={[{
                clickHandler: () => setModalOpen(false),
                label: 'Delete',
                variant: 'danger',
              }]}
              closeHandler={() => setModalOpen(false)}
              size={modalSize}
            >
              <p>
                Do you really want to delete the user <code>admin</code>?
                This cannot be undone.
              </p>
            </Modal>
          )}
        </div>
      </>
    );
  }}
</Playground>

On top of that, the modal is able to adjust to the width of its content.

üëâ Please note the auto width may not function correctly in combination with
other auto layout mechanisms, eg. the auto-width
[FormLayout](/components/layout/form-layout#label-width). It's just too much
magic that doesn't work together yet üé©.

<Playground>
  {() => {
    const [modalOpen, setModalOpen] = React.useState(false);
    return (
      <>
        <Button
          clickHandler={() => setModalOpen(true)}
          label="Launch auto-width modal"
          priority="outline"
        />
        <div>
          {modalOpen && (
            <Modal
              title="Delete the user?"
              actions={[{
                clickHandler: () => setModalOpen(false),
                label: 'Delete',
                variant: 'danger',
              }]}
              closeHandler={() => setModalOpen(false)}
              size="auto"
            >
              <p>
                Do you really want to delete the user <code>admin</code>?
                This cannot be undone.
              </p>
            </Modal>
          )}
        </div>
      </>
    );
  }}
</Playground>

## Scrolling Long Content

When modals become too long for the user's viewport or device, they scroll
independent of the page itself.

The inner implementation of body scrolling uses a partially
[customizable](#api) instance of the [ScrollView](/components/ui/scroll-view)
component.

<Playground>
  {() => {
    const [modalOpen, setModalOpen] = React.useState(false);
    const [modalScrollMode, setModalScrollMode] = React.useState('body');
    return (
      <>
        <Button
          clickHandler={() => {
            setModalScrollMode('body');
            setModalOpen(true);
          }}
          label="Launch modal with scrolling body"
          priority="outline"
        />
        <Button
          clickHandler={() => {
            setModalScrollMode('modal');
            setModalOpen(true);
          }}
          label="Launch scrolling modal"
          priority="outline"
        />
        <div>
          {modalOpen && (
            <Modal
              title="Modal with long content"
              actions={[{
                clickHandler: () => setModalOpen(false),
                label: 'OK',
              }]}
              autoFocus={false}
              closeHandler={() => setModalOpen(false)}
              scrollMode={modalScrollMode}
              size="small"
            >
              <p>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean
                commodo ligula eget dolor. Aenean massa.
              </p>
              <p>
                Cum sociis natoque penatibus et magnis dis parturient montes,
                nascetur ridiculus mus. Donec quam felis, ultricies nec,
                pellentesque eu, pretium quis, sem.
              </p>
              <p>
                Nulla consequat massa quis enim. Donec pede justo, fringilla
                vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus
                ut, imperdiet a, venenatis vitae, justo.
              </p>
              <p>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean
                commodo ligula eget dolor. Aenean massa.
              </p>
              <p>
                Cum sociis natoque penatibus et magnis dis parturient montes,
                nascetur ridiculus mus. Donec quam felis, ultricies nec,
                pellentesque eu, pretium quis, sem.
              </p>
              <p>
                Nulla consequat massa quis enim. Donec pede justo, fringilla
                vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus
                ut, imperdiet a, venenatis vitae, justo.
              </p>
              <p>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean
                commodo ligula eget dolor. Aenean massa.
              </p>
              <p>
                Cum sociis natoque penatibus et magnis dis parturient montes,
                nascetur ridiculus mus. Donec quam felis, ultricies nec,
                pellentesque eu, pretium quis, sem.
              </p>
              <p>
                Nulla consequat massa quis enim. Donec pede justo, fringilla
                vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus
                ut, imperdiet a, venenatis vitae, justo.
              </p>
              <p>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean
                commodo ligula eget dolor. Aenean massa.
              </p>
              <p>
                Cum sociis natoque penatibus et magnis dis parturient montes,
                nascetur ridiculus mus. Donec quam felis, ultricies nec,
                pellentesque eu, pretium quis, sem.
              </p>
              <p>
                Nulla consequat massa quis enim. Donec pede justo, fringilla
                vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus
                ut, imperdiet a, venenatis vitae, justo.
              </p>
              <p>
                Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean
                commodo ligula eget dolor. Aenean massa.
              </p>
              <p>
                Cum sociis natoque penatibus et magnis dis parturient montes,
                nascetur ridiculus mus. Donec quam felis, ultricies nec,
                pellentesque eu, pretium quis, sem.
              </p>
              <p>
                Nulla consequat massa quis enim. Donec pede justo, fringilla
                vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus
                ut, imperdiet a, venenatis vitae, justo.
              </p>
            </Modal>
          )}
        </div>
      </>
    );
  }}
</Playground>

### Long Content and Autofocus

üëâ If you set the `scrollMode` to `modal`, you may want to turn `autoFocus` off
to prevent the modal from scrolling to the end immediately after being opened.

## Blocking Modals

There are situations when you need to interrupt user's flow and lock entire UI
to prevent interaction. That's where blocking modals may come handy.

<Playground>
  {() => {
    const [modalOpen, setModalOpen] = React.useState(false);
    return (
      <>
        <Button
          clickHandler={() => setModalOpen(true)}
          label="Launch blocking modal"
          priority="outline"
        />
        <div>
          {modalOpen && (
            <Modal title="Update is in progress" size="auto">
              <p>
                <span role="img" aria-label="Warning">‚ö†Ô∏è</span>
                Please wait a few seconds until the firmware update is finished.
                Don&apos;t turn off the device!
              </p>
              <Button
                clickHandler={() => setModalOpen(false)}
                label="Close the demo"
                priority="flat"
              />
            </Modal>
          )}
        </div>
      </>
    );
  }}
</Playground>

## API

<Props of={ParsableModal} />

---

Next: [Radio ‚Üí](/components/ui/radio)
