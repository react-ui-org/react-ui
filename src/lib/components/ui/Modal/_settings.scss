@use '../../../styles/settings/borders';
@use '../../../styles/settings/typography';
@use '../../../styles/settings/z-indexes';
@use '../../../styles/tools/spacing';

$padding-x: spacing.of(5);
$padding-y: spacing.of(3);
$border-radius: borders.$radius;
$head-title-font-size: map-get(typography.$size-values, 1);
$z-index: z-indexes.$modal;
$overlay-z-index: z-indexes.$modal-overlay;
