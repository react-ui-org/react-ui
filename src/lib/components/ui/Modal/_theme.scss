$background: var(--rui-modal-background);
$box-shadow: var(--rui-modal-box-shadow);
$footer-background: var(--rui-modal-footer-background);
$overlay-background: var(--rui-modal-overlay-background);
$size-auto-min-width: var(--rui-modal-size-auto-min-width);
$size-auto-max-width: var(--rui-modal-size-auto-max-width);
$size-small-width: var(--rui-modal-size-small-width);
$size-medium-width: var(--rui-modal-size-medium-width);
$size-large-width: var(--rui-modal-size-large-width);
$size-fullscreen-height: var(--rui-modal-size-fullscreen-height);
$size-fullscreen-width: var(--rui-modal-size-fullscreen-width);
