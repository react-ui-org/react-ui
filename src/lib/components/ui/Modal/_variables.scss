@import '../../../styles/settings/borders';
@import '../../../styles/settings/colors';
@import '../../../styles/settings/typography';
@import '../../../styles/settings/zindexes';
@import '../../../styles/tools/spacing';

$modal-padding-x: spacing(5);
$modal-padding-y: spacing(3);
$modal-border-radius: $border-radius;
$modal-head-title-font-size: map-get($typography-size-values, 1);
$modal-zindex: $zindex-modal;
$modal-overlay-zindex: $zindex-modal-overlay;
