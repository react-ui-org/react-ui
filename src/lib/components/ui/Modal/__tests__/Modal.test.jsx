import React from 'react';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import Modal from '..';

describe('rendering', () => {
  it('renders correctly with mandatory props only', () => {
    const tree = mount((
      <Modal
        title="Modal title"
      >
        Modal content
      </Modal>
    ));

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with all props except translations', () => {
    const tree = mount((
      <Modal
        actions={[
          {
            clickHandler: () => {},
            label: 'Action',
          },
        ]}
        closeHandler={() => {}}
        id="custom-id"
        size="large"
        title="Modal title"
      >
        Modal content
      </Modal>
    ));

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with all props except translations and with a loading icon', () => {
    const tree = mount((
      <Modal
        actions={[
          {
            clickHandler: () => {},
            label: 'Action',
            loadingIcon: <span className="icon" />,
          },
        ]}
        closeHandler={() => {}}
        id="custom-id"
        size="small"
        title="Modal title"
      >
        Modal content
      </Modal>
    ));

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with translations', () => {
    const tree = mount((
      <Modal
        closeHandler={() => {}}
        title="Modal title"
        translations={{ close: 'Zavřít' }}
      >
        Modal content
      </Modal>
    ));

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls closeHandler() on Close button click', () => {
    const spy = sinon.spy();
    const component = mount((
      <Modal
        closeHandler={spy}
        title="Modal title"
      >
        Modal content
      </Modal>
    ));

    component.find('button').at(0).simulate('click');
    component.find('Button').at(0).simulate('click');
    expect(spy.calledTwice).toEqual(true);
  });
});

