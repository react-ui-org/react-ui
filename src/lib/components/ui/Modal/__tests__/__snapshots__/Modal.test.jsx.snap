// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with all props except translations 1`] = `
<WithTranslationContextComponent
  actions={
    Array [
      Object {
        "clickHandler": [Function],
        "label": "Action",
      },
    ]
  }
  closeHandler={[Function]}
  id="custom-id"
  size="large"
  title="Modal title"
  translations={null}
>
  <Modal
    actions={
      Array [
        Object {
          "clickHandler": [Function],
          "label": "Action",
        },
      ]
    }
    closeHandler={[Function]}
    id="custom-id"
    size="large"
    title="Modal title"
    translations={
      Object {
        "close": "Close",
      }
    }
  >
    <div
      className="overlay"
      id="custom-id"
      onClick={[Function]}
      onScroll={[Function]}
      role="presentation"
    >
      <div
        className="root
            isRootLarge"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="head"
        >
          <h3
            className="headTitle"
            id="custom-id__title"
          >
            Modal title
          </h3>
          <button
            className="close"
            id="custom-id__closeModalHeaderButton"
            onClick={[Function]}
            title="Close"
            type="button"
          >
            ×
          </button>
        </div>
        <div
          className="body"
          id="custom-id__content"
        >
          Modal content
        </div>
        <div
          className="footer"
        >
          <ForwardRef(withForwardedRef(Button))
            clickHandler={[Function]}
            key="Action"
            label="Action"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={false}
              clickHandler={[Function]}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              label="Action"
              labelVisibility="all"
              loadingIcon={null}
              priority="default"
              size="medium"
              startCorner={null}
              type="button"
              variant="primary"
            >
              <button
                className="root
        priorityDefault
        sizeMedium
        variantPrimary"
                disabled={false}
                onClick={[Function]}
                title={null}
                type="button"
              >
                <span
                  className="label"
                >
                  Action
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
          <ForwardRef(withForwardedRef(Button))
            clickHandler={[Function]}
            id="custom-id__closeModalFooterButton"
            label="Close"
            priority="flat"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={false}
              clickHandler={[Function]}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              id="custom-id__closeModalFooterButton"
              label="Close"
              labelVisibility="all"
              loadingIcon={null}
              priority="flat"
              size="medium"
              startCorner={null}
              type="button"
              variant="primary"
            >
              <button
                className="root
        priorityFlat
        sizeMedium
        variantPrimary"
                disabled={false}
                id="custom-id__closeModalFooterButton"
                onClick={[Function]}
                title={null}
                type="button"
              >
                <span
                  className="label"
                  id="custom-id__closeModalFooterButton__labelText"
                >
                  Close
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </div>
      </div>
    </div>
  </Modal>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with all props except translations and with a loading icon 1`] = `
<WithTranslationContextComponent
  actions={
    Array [
      Object {
        "clickHandler": [Function],
        "label": "Action",
        "loadingIcon": <span
          className="icon"
        />,
      },
    ]
  }
  closeHandler={[Function]}
  id="custom-id"
  size="small"
  title="Modal title"
  translations={null}
>
  <Modal
    actions={
      Array [
        Object {
          "clickHandler": [Function],
          "label": "Action",
          "loadingIcon": <span
            className="icon"
          />,
        },
      ]
    }
    closeHandler={[Function]}
    id="custom-id"
    size="small"
    title="Modal title"
    translations={
      Object {
        "close": "Close",
      }
    }
  >
    <div
      className="overlay"
      id="custom-id"
      onClick={[Function]}
      onScroll={[Function]}
      role="presentation"
    >
      <div
        className="root
            isRootSmall"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="head"
        >
          <h3
            className="headTitle"
            id="custom-id__title"
          >
            Modal title
          </h3>
          <button
            className="close"
            id="custom-id__closeModalHeaderButton"
            onClick={[Function]}
            title="Close"
            type="button"
          >
            ×
          </button>
        </div>
        <div
          className="body"
          id="custom-id__content"
        >
          Modal content
        </div>
        <div
          className="footer"
        >
          <ForwardRef(withForwardedRef(Button))
            clickHandler={[Function]}
            key="Action"
            label="Action"
            loadingIcon={
              <span
                className="icon"
              />
            }
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={false}
              clickHandler={[Function]}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              label="Action"
              labelVisibility="all"
              loadingIcon={
                <span
                  className="icon"
                />
              }
              priority="default"
              size="medium"
              startCorner={null}
              type="button"
              variant="primary"
            >
              <button
                className="root
        priorityDefault
        sizeMedium
        variantPrimary"
                disabled={true}
                onClick={[Function]}
                title={null}
                type="button"
              >
                <span
                  className="label"
                >
                  Action
                </span>
                <span
                  className="loadingIcon"
                >
                  <span
                    className="icon"
                  />
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
          <ForwardRef(withForwardedRef(Button))
            clickHandler={[Function]}
            id="custom-id__closeModalFooterButton"
            label="Close"
            priority="flat"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={false}
              clickHandler={[Function]}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              id="custom-id__closeModalFooterButton"
              label="Close"
              labelVisibility="all"
              loadingIcon={null}
              priority="flat"
              size="medium"
              startCorner={null}
              type="button"
              variant="primary"
            >
              <button
                className="root
        priorityFlat
        sizeMedium
        variantPrimary"
                disabled={false}
                id="custom-id__closeModalFooterButton"
                onClick={[Function]}
                title={null}
                type="button"
              >
                <span
                  className="label"
                  id="custom-id__closeModalFooterButton__labelText"
                >
                  Close
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </div>
      </div>
    </div>
  </Modal>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with mandatory props only 1`] = `
<WithTranslationContextComponent
  title="Modal title"
  translations={null}
>
  <Modal
    actions={Array []}
    closeHandler={null}
    size="medium"
    title="Modal title"
    translations={
      Object {
        "close": "Close",
      }
    }
  >
    <div
      className="overlay"
      onClick={[Function]}
      onScroll={[Function]}
      role="presentation"
    >
      <div
        className="root
            isRootMedium"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="head"
        >
          <h3
            className="headTitle"
          >
            Modal title
          </h3>
        </div>
        <div
          className="body"
        >
          Modal content
        </div>
        <div
          className="footer"
        />
      </div>
    </div>
  </Modal>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with translations 1`] = `
<WithTranslationContextComponent
  closeHandler={[Function]}
  title="Modal title"
  translations={
    Object {
      "close": "Zavřít",
    }
  }
>
  <Modal
    actions={Array []}
    closeHandler={[Function]}
    size="medium"
    title="Modal title"
    translations={
      Object {
        "close": "Zavřít",
      }
    }
  >
    <div
      className="overlay"
      onClick={[Function]}
      onScroll={[Function]}
      role="presentation"
    >
      <div
        className="root
            isRootMedium"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="head"
        >
          <h3
            className="headTitle"
          >
            Modal title
          </h3>
          <button
            className="close"
            onClick={[Function]}
            title="Zavřít"
            type="button"
          >
            ×
          </button>
        </div>
        <div
          className="body"
        >
          Modal content
        </div>
        <div
          className="footer"
        >
          <ForwardRef(withForwardedRef(Button))
            clickHandler={[Function]}
            label="Zavřít"
            priority="flat"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={false}
              clickHandler={[Function]}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              label="Zavřít"
              labelVisibility="all"
              loadingIcon={null}
              priority="flat"
              size="medium"
              startCorner={null}
              type="button"
              variant="primary"
            >
              <button
                className="root
        priorityFlat
        sizeMedium
        variantPrimary"
                disabled={false}
                onClick={[Function]}
                title={null}
                type="button"
              >
                <span
                  className="label"
                >
                  Zavřít
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </div>
      </div>
    </div>
  </Modal>
</WithTranslationContextComponent>
`;
